controllers/provisioning_controller.go

```diff
@@ -650,14 +654,45 @@ func (r *ProvisioningReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	klog.InfoS("Network stack calculation", "NetworkStack", r.NetworkStack)
 
+	provisioningFilter := predicate.NewPredicateFuncs(func(object client.Object) bool {
+		if object.GetName() == metal3iov1alpha1.ProvisioningSingletonName {
+			return true
+		}
+		klog.Info("ignoring provisioning config with unexpected name", "name", object.GetName(), "expected-name", metal3iov1alpha1.ProvisioningSingletonName)
+		return false
+	})
+
+	// Watch Secret openshift-config/pull-secret. If this secret changes, we must requeue the provisioning Singleton,
+	// if it exists.
+	watchOCPConfigPullSecret := func(ctx context.Context, object client.Object) []reconcile.Request {
+		prov, err := r.readProvisioningCR(ctx)
+		// readProvisioningCR can return nil, nil upon IsNotFound, account for this as well.
+		if err != nil || prov == nil {
+			return []reconcile.Request{}
+		}
+		return []reconcile.Request{
+			{
+				NamespacedName: types.NamespacedName{
+					Namespace: prov.Namespace,
+					Name:      prov.Name,
+				},
+			},
+		}
+	}
+	secretFilter := predicate.NewPredicateFuncs(func(object client.Object) bool {
+		return object.GetNamespace() == provisioning.OpenshiftConfigNamespace &&
+			object.GetName() == provisioning.PullSecretName
+	})
+
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&metal3iov1alpha1.Provisioning{}).
+		For(&metal3iov1alpha1.Provisioning{}, builder.WithPredicates(provisioningFilter)).
```

Comment:
This is not needed.

It was added in https://github.com/openshift/cluster-baremetal-operator/pull/351/commits/d4e709bbfbae6d316f2e76bec18b0e10a45ac93e to cancel out the effect of using `handler.EnqueueRequestForObject{}` (instead of `handler.EnqueueRequestFromMapFunc`) on resources with different names to the provisioning CR. We don't need to re-add it.